src/dbnode/storage/mediator_test.go

```diff
@@ -47,11 +50,28 @@ func TestDatabaseMediatorOpenClose(t *testing.T) {
 	m, err := newMediator(db, nil, opts)
 	require.NoError(t, err)
 
+	var executed atomic.Bool
+
+	backgroundProcess := NewMockBackgroundProcess(ctrl)
+	gomock.InOrder(
+		backgroundProcess.EXPECT().Run().Do(func() {
+			executed.Store(true)
+		}),
+		backgroundProcess.EXPECT().Stop(),
+	)
+
+	m.RegisterBackgroundProcess(backgroundProcess)
+
 	require.Equal(t, errMediatorNotOpen, m.Close())
 
 	require.NoError(t, m.Open())
 	require.Equal(t, errMediatorAlreadyOpen, m.Open())
 
+	hasExecuted := xclock.WaitUntil(func() bool {
+		return executed.Load()
+	}, time.Second)
```

Comment:
Should we set `InstrumentOptions().ReportInterval()` in this test? Currently the default reporting interval is 1 sec which may *potentially* miss this? May be better to make the interval shorter for this test?

Reply:
👍